src/test/java/com/google/vinet/data/MatchingAlgorithmTest.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.data;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.time.temporal.ChronoUnit.HOURS;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class MatchingAlgorithmTest {
```

Comment:
Complex cases of the algorithm hasn't been checked.

Reply:
I have added more tests with larger sample sets, and some overlapping ones. Would that cover the complex cases, or would more conflicting timeslots do it?

Reply:
Have one with bit more conflicting timeslots.